### PR TITLE
fix(proxy): urn warnings for upgrade

### PR DESCRIPTION
**Pull request by FintanH**
Fixes https://github.com/radicle-dev/radicle-upstream/issues/1560

For the existing remote, I realised that the URL won't parse full-stop. So I decided to handle this case specifically by renaming the existing remote, thus preserving the reference data. Then we create a fresh `rad` remote. A user can then copy any of their references over from `temp_rad` and remove it when they're done.

~For the search box, I'm not sure how to detect if it was an old URN and give a specific message in that case. So what I did here was give a small prompt that it might be an old URN. Maybe we could try document the differences in the community post.~

### CODE REVIEW

File: ui/src/search.ts [hunk 1]
```diff
@@ -47,7 +47,7 @@ const VALID_URN_MATCH = /^rad:git:[1-9A-HJ-NP-Za-km-z]{37}/;
 const urnConstraints = {
   format: {
     pattern: VALID_URN_MATCH,
-    message: `Not a valid project URN`,
+    message: `Not a valid project URN, perhaps you're using an old URN?`,
```
Comment (cloudhead) [inline ui/src/search.ts:50]: I'm not sure this is an improvement, as-is. Users will have no idea what that means.

File: proxy/coco/src/project/create/validation.rs [hunk 2]
```diff
@@ -375,6 +375,14 @@ impl Repository {
         url: &LocalUrl,
     ) -> Result<Option<Remote<LocalUrl>>, Error> {
         match Remote::<LocalUrl>::find(repo, reflike!("rad")) {
+            Err(remote::FindError::ParseUrl(_)) => {
+                log::warn!("an old/invalid URL was found when trying to load the `rad` remote");
+                log::warn!(
+                    "we are going to rename the remote to `temp_rad` and create a new `rad` remote"
+                );
+                repo.remote_rename("rad", "temp_rad")?;
```
Comment (cloudhead) [inline proxy/coco/src/project/create/validation.rs:383]: It's not temporary is it? Perhaps `rad_old`?

File: ui/src/search.ts (hunk 1 above)
Comment (FintanH) [reply ui/src/search.ts:50]: > Users will have no idea what that means.

Not until they read the post ;P

How could this be improved? A link to the post (possibly to a specific URN section)?

File: proxy/coco/src/project/create/validation.rs (hunk 2 above)
Comment (FintanH) [reply proxy/coco/src/project/create/validation.rs:383]: It is if they remove it later :smile: `rad_old` works too :+1:

File: ui/src/search.ts (hunk 1 above)
Comment (cloudhead) [reply ui/src/search.ts:50]: Hmm, I wonder if it's worthwhile to have this (to then have to eventually remove it). I think just leaving it as it was might be fine -- we can explain in the post that old URNs aren't compatible, at least they will have the context there.

Comment (FintanH) [reply ui/src/search.ts:50]: Works for me. We can also add a section to the troubleshooting docs

**Review by cloudhead — APPROVED** [review]

**Pull request merged**